tidy3d/components/simulation.py

```diff
@@ -1784,7 +1784,7 @@ def _get_structures_plane(
 
     @staticmethod
     def _filter_structures_plane(  # pylint:disable=too-many-locals
-        structures: List[Structure], x: float = None, y: float = None, z: float = None
+        structures: List[Structure], plane: Box
```

Comment:
I think we need to ensure that the plane is in fact 2D and error if not, especially in the public facing methods.